forge/ee/routes/catalogues/index.js

```diff
@@ -62,6 +65,92 @@ module.exports = async function (app) {
         }
     })
 
+    /**
+     * Uploads a NPM packaged subflow to teams private repo
+     * @name /api/v1/teams/:teamId/npm/subflow
+     * @static
+     * @memberof forge.routes.api.team.npm
+     */
+    app.put('/npm/subflow', {
+        preHandler: async (request, reply) => {
+            if (request.session.ownerType !== 'project') {
+                reply.code(401).send({ code: 'unauthorized', error: 'unauthorized' })
+            }
+            const project = await app.db.models.Project.byId(request.session.ownerId)
+            if (request.params.teamId !== project.Team.hashid) {
+                // AccesToken being used - but not owned by project in this team
+                reply.code(404).send({ code: 'not_found', error: 'Not Found' })
+                return // eslint-disable-line no-useless-return
+            }
+        },
+        schema: {
+            summary: 'Allows Subflow packages to be stored in the Team Registry',
+            tags: ['NPM Package'],
+            body: {
+                type: 'object',
+                properties: {
+                    package: {
+                        type: 'object',
+                        additionalProperties: true
+                    },
```

Comment:
Since name and version are require props, we should let swagger take care of checking and handing this:


```suggestion
                    package: {
                        type: 'object',
                        properties: {
                            name: { type: 'string' },
                            version: { type: 'string' }
                        },
                        required: ['name', 'version'],
                        additionalProperties: true
                    },
```